### PR TITLE
chore: bump node from 14 to 16

### PR DESCRIPTION
**Pull request by millsp**

### CODE REVIEW

File: platforms-serverless/lambda/update-config.sh [hunk 1]
```diff
@@ -2,4 +2,4 @@
 
 set -eux
 
-aws lambda update-function-configuration --function-name prisma2-e2e-tests --runtime nodejs14.x --handler index.handler --timeout 10
+aws lambda update-function-configuration --function-name prisma2-e2e-tests --runtime 6.x --handler index.handler --timeout 10
```
Comment (janpio) [inline platforms-serverless/lambda/update-config.sh]:
```suggestion
aws lambda update-function-configuration --function-name prisma2-e2e-tests --runtime nodejs16.x --handler index.handler --timeout 10
```

**Review by Jolg42 — APPROVED** [review]
Looks good, once merged we should check all functions from https://vercel.com/prisma-orm and switch them to 16

**Comment by millsp** [comment]
(After reviving the PR, new tests are failing).

**Comment by Jolg42** [comment]
Note:
I updated pkg version because I think the previous version didn't support Node 16
I updated Vercel Node version in the admin UI
For Netlify I changed the "NODE_VERSION" env var to 16 in https://app.netlify.com/sites/prisma-ecosystem-tests-netlify-github/configuration/env#NODE_VERSION

**Comment by millsp** [comment]
I had to revert the use of pnpm@8 because some platforms are still on Node v16.10 or 16.4, but pnpm@8 requires at least Node v16.14. Among these, I remember Amazon EC2, Netlify, and Macstadium, maybe in a couple of months we can try again, or are all of these configurable wrt thei Node version?

**Comment by millsp** [comment]
Also, you might notice this `4.17.0-integration-fix-internals-vercel-pkg-win-resolve-binary.5`
 version is in use. This fixes problems with `prisma` and Vercel's `pkg`, PR [here](https://github.com/prisma/prisma/pull/20072). We'll need to merged that first.

**Comment by millsp** [comment]
I did another minor change, which was to change `gcp-functions` to use a `package-lock.json` instead of a `pnpm-lock.yaml`. Google Cloud Functions didn't support `pnpm` in the past but we still had a `pnpm` lockfile for some reason (maybe a small mistake while porting the repo from `yarn` to `pnpm`), and it seems this is not the case since the switch to NodeJS 16, they are to be able to handle `pnpm`. However, they seem to be using another version than pnpm@7, so we get errors. I didn't want to go through the hurdle and switched to `npm`, as it should have been, and like for the very few projects that do need it.

File: .github/prisma-version.txt [hunk 2]
```diff
@@ -1 +1 @@
-4.17.0-dev.60
+4.17.0-integration-fix-internals-vercel-pkg-win-resolve-binary.5
```
Comment (janpio) [inline .github/prisma-version.txt]: To be fixed before merge

File: platforms/aws-graviton/run.sh [hunk 3]
```diff
@@ -21,7 +21,7 @@ else
       cd /home/ec2-user/aws-graviton/$GITHUB_JOB/$GITHUB_RUN_ID/$PRISMA_CLIENT_ENGINE_TYPE;
       export PRISMA_CLIENT_ENGINE_TYPE=\"library\"
       rm -rf ./node_modules;
-      npm i -g pnpm@8;
+      npm i -g pnpm@7;
```
Comment (Jolg42) [inline platforms/aws-graviton/run.sh:24]: pnpm v8 should work there? We can check in a separate PR for sure

**Review by Jolg42 — APPROVED** [review]
:shipit:

**Pull request merged**